src/Lib.P2P/Multiplex/Muxer.cs

```diff
@@ -0,0 +1,340 @@
+                            if (substream != null)
+                            {
+                                _log.Warn($"Stream {substream.Id} already exists");
+                                continue;
+                            }
+
+                            substream = new Substream
+                            {
+                                Id = header.StreamId,
+                                Name = Encoding.UTF8.GetString(payload),
+                                Muxer = this
+                            };
+                            if (!Substreams.TryAdd(substream.Id, substream))
+
+                                // Should not happen.
+                                throw new Exception($"Stream {substream.Id} already exists");
+                            SubstreamCreated?.Invoke(this, substream);
+
+                            // Special hack for go-ipfs
+#if true
+                            if (Receiver && (substream.Id & 1) == 1)
+                            {
+                                _log.Debug($"go-hack sending newstream {substream.Id}");
+                                using (await AcquireWriteAccessAsync().ConfigureAwait(false))
+                                {
+                                    var hdr = new Header
+                                    {
+                                        StreamId = substream.Id,
+                                        PacketType = PacketType.NewStream
+                                    };
+                                    await hdr.WriteAsync(Channel, cancel).ConfigureAwait(false);
+                                    Channel.WriteByte(0); // length
+                                    await Channel.FlushAsync(cancel).ConfigureAwait(false);
+                                }
+                            }
+#endif
+                            break;
+
+                        case PacketType.MessageInitiator:
+                            if (substream == null)
+                            {
+                                _log.Warn($"Message to unknown stream #{header.StreamId}");
+                                continue;
+                            }
+
+                            substream.AddData(payload);
+                            break;
+
+                        case PacketType.MessageReceiver:
+                            if (substream == null)
+                            {
+                                _log.Warn($"Message to unknown stream #{header.StreamId}");
+                                continue;
+                            }
+
+                            substream.AddData(payload);
+                            break;
+
+                        case PacketType.CloseInitiator:
+                        case PacketType.CloseReceiver:
+                        case PacketType.ResetInitiator:
+                        case PacketType.ResetReceiver:
+                            if (substream == null)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this code to not nest more than 3 control flow statements.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)